dsp.js

```
@@ -582,7 +583,7 @@ function RFFT(bufferSize, sampleRate) {
       i++;
 
       h = halfSize << 1;
-      while (((h = h >> 1), !((r ^= h) & h)));
+      while (((h = h >> 1), !((r ^= h) & h))){}
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)